modules/oci-build/01_mod.mk

```diff
@@ -0,0 +1,198 @@
+endif
+ifeq ($(go_$1_main_dir:%/=/),/)
+$$(error go_$1_main_dir "$(go_$1_main_dir)" should be a directory path that DOES NOT end with "/")
+endif
+ifeq ($(go_$1_main_dir:%.go=.go),.go)
+$$(error go_$1_main_dir "$(go_$1_main_dir)" should be a directory path that DOES NOT end with ".go")
+endif
+ifneq ($(go_$1_mod_dir:.%=.),.)
+$$(error go_$1_mod_dir "$(go_$1_mod_dir)" should be a directory path that DOES start with ".")
+endif
+ifeq ($(go_$1_mod_dir:%/=/),/)
+$$(error go_$1_mod_dir "$(go_$1_mod_dir)" should be a directory path that DOES NOT end with "/")
+endif
+ifeq ($(go_$1_mod_dir:%.go=.go),.go)
+$$(error go_$1_mod_dir "$(go_$1_mod_dir)" should be a directory path that DOES NOT end with ".go")
+endif
+ifeq ($(wildcard $(go_$1_mod_dir)/go.mod),)
+$$(error go_$1_mod_dir "$(go_$1_mod_dir)" does not contain a go.mod file)
+endif
+ifeq ($(wildcard $(go_$1_mod_dir)/$(go_$1_main_dir)/main.go),)
+$$(error go_$1_main_dir "$(go_$1_mod_dir)" does not contain a main.go file)
+endif
+
+# Validate the config required to build OCI images
+ifneq ($(words $(oci_$1_image_name_development)),1)
+$$(error oci_$1_image_name_development "$(oci_$1_image_name_development)" should be a single image name)
+endif
+
+endef
+
+$(foreach build_name,$(build_names),$(eval $(call check_per_build_variables,$(build_name))))
+
+# Create variables holding targets
+#
+# We create the following targets for each $(build_names)
+# - oci-build-$(build_name) = build the oci directory
+# - oci-load-$(build_name) = load the image into docker using the oci_$(build_name)_image_name_development variable
+# - docker-tarball-$(build_name) = build a "docker load" compatible tarball of the image
+# - ko-config-$(build_name) = generate "ko" config for a given build
+oci_build_targets := $(build_names:%=oci-build-%)
+oci_load_targets := $(build_names:%=oci-load-%)
+docker_tarball_targets := $(build_names:%=docker-tarball-%)
+ko_config_targets := $(build_names:%=ko-config-%)
+
+# Derive config based on user config
+# 
+# - oci_layout_path_$(build_name) = path that the OCI image will be saved in OCI layout directory format
+# - oci_digest_path_$(build_name) = path to the file that will contain the digests
+# - ko_config_path_$(build_name) = path to the ko config file
+# - docker_tarball_path_$(build_name) = path that the docker tarball that the docker-tarball-$(build_name) will produce
+$(foreach build_name,$(build_names),$(eval oci_layout_path_$(build_name) := $(bin_dir)/scratch/image/oci-layout-$(build_name).$(oci_$(build_name)_image_tag)))
+$(foreach build_name,$(build_names),$(eval oci_digest_path_$(build_name) := $(CURDIR)/$(oci_layout_path_$(build_name)).digests))
+$(foreach build_name,$(build_names),$(eval ko_config_path_$(build_name) := $(CURDIR)/$(oci_layout_path_$(build_name)).ko_config.yaml))
+$(foreach build_name,$(build_names),$(eval docker_tarball_path_$(build_name) := $(CURDIR)/$(oci_layout_path_$(build_name)).docker.tar))
```

Comment:
I think we can move these definitions to a 00_mod.mk file and move the oci-publish/02_mod.mk file to oci-publish/01_mod.mk

Reply:
Yes, we can move everything. All these variables are expected to be defined in the make/00_mod.mk file anyways.